global.backend/session_operation.py

```diff
@@ -0,0 +1,174 @@
+""" Definition of 2 classes:
+    1. Session: Object used to store the session string retrieved from crawler server.
+    2. SessionCache: Container that cache the session by visa_type and places, along with
+        method for manipulating the session.
+
+    Refactored out from ../global/crawler/lite_visa.py
+"""
+import os
+import json
+import random
+import string
+import logging
+from queue import Queue
+from typing import Optional
+from collections import defaultdict
+
+import global_var as G
+
+
+class Session:
+    """ Store the session from CGI/AIS systems.
+        AIS system session has a variable `schedule_id` coming along, in order to
+        align the attributes of session objects from two systems, the field with be
+        None for session instances of CGI.
```

Comment:
> Can we append this schedule_id inside the session, i.e., session_ais = (session, schedule_id) instead of a string?

Do you mean the `session` argument take both `str` and `tuple` for input?

> 事实上还有第三个签证系统 https://evisaforms.state.gov/Instructions/SchedulingSystem.asp 这是最老的，如果后面要加这个的话，你这个代码好多都得改……？

是因为session的形式不一样吗

Reply:
> Do you mean the session argument take both str and tuple for input?

Yes

> 是因为session的形式不一样吗

我还没看但……多半不一样